cEP-0024.md

```diff
@@ -0,0 +1,96 @@
+our community, for example automatically close PRs from master. Many of them
+are general plugins, which are required by every org, but we need more plugins
+which support coala development workflow.
+
+## New Features
+
+### Auto-assign issue based on commit mention in a PR
+
+When a PR is opened in the repo and if any of it's commit
+mentions(fixes, closes) a unassigned issues, auto assign commit author to the
+mentioned unassigned issues.
+
+### Auto-reject PR opened on master of forks
+
+Normally, contributors should make changes in different branch from master of
+forks, and submit a PR. But, sometimes we receive PRs from master of forks.
+To solve this problem, we need a mechanism by which gitmate automatically close
+those PR’s, and comment on the PR informing the PR author to work on a
+different branch, comment message will be configurable.
+
+### Auto-mark PR with necessary labels as per the reviews done
+
+When a reviewer reviews a PR, he either approves changes or request changes or
+comments something and leaves it there, but the PR would be still in the review
+queue, so there should be an automation to mark approve label if reviewer
+approves changes and mark WIP if reviewer requests changes or comments on the
+source code when there is CI failure. This will be a part of ack plugin.
+A setting to limit this plugin to certain teams will be added.
+
+### Fixes vs Closes automated checking
+
+Our members often confuse when we should use `Fixes` or `Closes` in a commit
+message. So, GitMate should automatically check every commit for issues which
+is being Fixed or Closed, if those issues have bug label (configurable),
+`Fixes` should be used, and vice versa. If there is an error, GitMate will
+comment on the PR with a configurable message and remove 'pending review'
+labels, mark 'wip' label, this will be part of ack plugin.
+
+### Assign Plugin
+
+Assign issue when a user comments on an issue requesting for an `assignment`
+to that issue. Will support settings for each team like limit team to work on
+issues with specific set of labels, limit number of issues of specified label,
+a team member can work on, block assignment for certain labelled issues. corobo
+does all these things, but it is based just on gitter messages and corobo is
+used in coala only, but gitmate is based on github, used by many other users
+and is more configurable.
+
+```
+Settings:
+
+Blocked issues label: status/blocked
+
+Team - newcomer
+Label - newcomer  Limit - 1
+Label - low       Limit - 2
+
+Team 2...
+```
+
+```
+> @gitmate-bot assign
+
+  @user, you have been assigned
+```
+
+### Auto-review merge conflicts
+
+Check for merge conflicts in a PR, if there are any conflicts, mark the PR with
+a certain label (configurable) and post a message (configurable) as a comment
+on the PR. Use GitHub API to know if the the PR has merge conflicts by
+`mergeable_state` property in pull request. As Gitlab doesn't have this API,
```

Comment:
Is `mergeable_state` supported by IGitt ?

Will the GitLab implementation be done transparently in IGitt ?

Reply:
I will add `mergeable_state` in IGitt. GitLab implementation will be done in gitmate as per our needs.

Reply:
Hi, there is  ``"merge_status": "can_be_merged"`` key, value in Merge Request API of GitLab: https://docs.gitlab.com/ee/api/merge_requests.html

You should briefly discuss why are we not using it and if something else can be used in the future.  

Also, please check the following: 
- https://gitlab.com/gitlab-org/gitlab-ce/issues/42344
- https://gitlab.com/gitlab-org/gitlab-ce/issues/21829
- https://gitlab.com/gitlab-org/gitlab-ee/blob/master/app/models/merge_request.rb#L684
- https://gitlab.com/gitlab-org/gitlab-ee/issues/3169

If we can find any solution using the API itself, it would be much better than using the docker container.

Reply:
Also, @Vamshi99, I'd rather suggest using a unified approach to building this API into IGitt for both GitHub and GitLab, i.e. either use the API for both GitHub and GitLab or use containerized approach for both. 

If it is possible to infer data via API and analyze the mergeability status, it'd be a lot easier to work with. 

If that's not the case, since, IGitt cannot (and maybe, should not) spawn new containers to do its job. This functionality should be baked into gitmate-2, rather than IGitt, if it involves fiddling with docker.